pyo3-ffi/src/object.rs

```diff
@@ -28,21 +39,39 @@ pub const PyObject_HEAD_INIT: PyObject = PyObject {
 // skipped PyObject_VAR_HEAD
 // skipped Py_INVALID_SIZE
 
+#[repr(C)]
+#[derive(Copy, Clone)]
+#[cfg(Py_3_12)]
+pub union __pyo3_object_ob_refcnt {
+    pub ob_refcnt: Py_ssize_t,
+    #[cfg(not(target_pointer_width = "32"))]
```

Comment:
I think this only works for `target_pointer_width = "64"` and would hence would prefer that as the `cfg` attribute here and above. (I don't think we can reasonably CPython on CHERI here in any case.)

Reply:
The C header used `# if SIZEOF_VOID_P > 4`, of which the closest analogy I could find was `not(target_pointer_width = "32")`, disregarding 16 bits which I don't think we would try to support.

Reply:
https://github.com/python/cpython/blob/e3b5ed7b1cb8ad12aa285cf78303c6dda0953181/Include/object.h#L182-L184

Reply:
I think copying verbatim is not useful here and would prefer the direct formulation as the structure does not make sense for anything but 64 bit architectures.